service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -106,13 +106,29 @@ public String getWsmServiceAccountToken() throws IOException {
    * @param userRequest - request object for this user
    * @return - email address of user
    */
-  public String getRequestUserEmail(AuthenticatedUserRequest userRequest)
+  public String getEmailFromRequestOrSam(AuthenticatedUserRequest userRequest)
       throws InterruptedException {
     Optional<String> emailMaybe = Optional.ofNullable(userRequest.getEmail());
```

Comment:
Can we (in a separate PR) make this getter return `Optional<String>`? Same question for the subject ID.

Reply:
Tracking in PF-1001. Agreed that should be a separate PR, I think it would have a lot of small changes scattered across the codebase